tiledb/sm/array_schema/domain.cc

```diff
@@ -659,12 +659,18 @@ bool Domain::overlap(const NDRange& r1, const NDRange& r2) const {
   return true;
 }
 
-double Domain::overlap_ratio(const NDRange& r1, const NDRange& r2) const {
+double Domain::overlap_ratio(
+    const NDRange& r1,
+    const std::vector<bool>& r1_default,
+    const NDRange& r2) const {
   double ratio = 1.0;
   assert(dim_num_ == r1.size());
   assert(dim_num_ == r2.size());
 
   for (unsigned d = 0; d < dim_num_; ++d) {
+    if (r1_default[d])
+      continue;
```

Comment:
Appears this will now exclude a 'default' range from consideration of any overlap and in calculating ratio, 
is that desireable?

Reply:
Yes, in this case, if it's default, it means that we include everything in this dimension, which means overlap ration should be 1, which would leave ratio unaltered.